### DISCUSSION TITLE
What metadata do I need?

### DISCUSSION THREAD

**Discussion by charr238** [CollectionBuilder/collectionbuilder.github.io#119] [Q&A]
I am looking to create a personal repository of all photos and videos I have taken over the years. It is certainly a daunting task, but I think it will be well worth it. 

I am looking at about 5000 items that I'd like to work through. I am having some issues understanding the CB page describing the required metadata. Here is what I believe is all that I need to get started:

objectid | filename | title | creator | date | description | subject | location | type | format | language

**Am I missing any?**

**Comment by evanwill** [comment]
@charr238 yep, those basics will work! That looks like the general template that most people get started with. 

The absolute minimum metadata required is "objectid" and "title" (i.e. the site will build and some items will show up) . 
The other fields will allow other things to show up and function, as outlined in the [CB metadata docs section](https://collectionbuilder.github.io/cb-docs/docs/metadata/gh_metadata/#required-fields-for-collectionbuilder-gh-and-sheets). 
So the next sort of required fields are:

- filename --> will get the image (or other item type) to show up
- format --> is necessary for display logic in the template, if you don't have it, the Items will be displayed incorrectly since the template doesn't know what type of item it is. 

After that the other fields are optional, but if you use the ones described in the docs, they will work with the default configuration of the template. So date, subject, location are the next level of suggested fields, since they are what the default visualizations are looking for.

Next, the optional fields creator, description, source are typically used because people would like to know that information--but they aren't immediately needed by anything in the template. And you can add what ever other fields you would like to display!
